google/cloud/bigquery/model.py

```diff
@@ -66,140 +62,122 @@ def __init__(self, model_ref):
             model_ref = ModelReference.from_string(model_ref)
 
         if model_ref:
-            self._proto.model_reference.CopyFrom(model_ref._proto)
+            self._properties["modelReference"] = model_ref.to_api_repr()
 
     @property
-    def reference(self):
-        """A :class:`~google.cloud.bigquery.model.ModelReference` pointing to
-        this model.
+    def reference(self) -> Optional["ModelReference"]:
+        """A model reference pointing to this model.
 
         Read-only.
-
-        Returns:
-            google.cloud.bigquery.model.ModelReference: pointer to this model.
         """
-        ref = ModelReference()
-        ref._proto = self._proto.model_reference
-        return ref
+        resource = self._properties.get("modelReference")
+        if resource is not None:
+            return ModelReference.from_api_repr(resource)
 
     @property
-    def project(self):
-        """str: Project bound to the model"""
+    def project(self) -> str:
+        """Project bound to the model."""
         return self.reference.project
 
     @property
-    def dataset_id(self):
-        """str: ID of dataset containing the model."""
+    def dataset_id(self) -> str:
+        """ID of dataset containing the model."""
         return self.reference.dataset_id
 
     @property
-    def model_id(self):
-        """str: The model ID."""
+    def model_id(self) -> str:
+        """The model ID."""
         return self.reference.model_id
 
     @property
-    def path(self):
-        """str: URL path for the model's APIs."""
+    def path(self) -> str:
+        """URL path for the model's APIs."""
         return self.reference.path
 
     @property
-    def location(self):
-        """str: The geographic location where the model resides. This value
-        is inherited from the dataset.
+    def location(self) -> str:
+        """The geographic location where the model resides.
+
+        This value is inherited from the dataset.
 
         Read-only.
         """
-        return self._proto.location
+        return self._properties.get("location")
 
     @property
-    def etag(self):
-        """str: ETag for the model resource (:data:`None` until
-        set from the server).
+    def etag(self) -> str:
+        """ETag for the model resource (:data:`None` until set from the server).
 
         Read-only.
         """
-        return self._proto.etag
+        return self._properties.get("etag")
 
     @property
-    def created(self):
-        """Union[datetime.datetime, None]: Datetime at which the model was
-        created (:data:`None` until set from the server).
+    def created(self) -> Optional[datetime.datetime]:
+        """Datetime at which the model was created (:data:`None` until set from the server).
 
         Read-only.
         """
-        value = self._proto.creation_time
+        value = self._properties.get("creationTime")
         if value is not None and value != 0:
```

Comment:
I don't think we need to check for `value != 0` anymore. I believe we added this a side-effect of protocol buffers v3 behavior with optional fields -- they'd always return the default value.